lib/views/users/users_page.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:flutter/material.dart';
+import 'package:ibenta_technical_test_flutter/data/api_client.dart';
+import 'package:ibenta_technical_test_flutter/data/api_constants.dart';
+import 'package:ibenta_technical_test_flutter/data/users_repository.dart';
+import 'package:ibenta_technical_test_flutter/models/user.dart';
+import 'package:ibenta_technical_test_flutter/views/base_page.dart';
+import 'package:ibenta_technical_test_flutter/views/users/user_page.dart';
+import 'package:oauth2/oauth2.dart' as oauth2;
+
+class UsersPage extends StatefulWidget {
+  UsersPage({Key? key}) : super(key: key);
+
+  @override
+  _UsersPageState createState() => _UsersPageState();
+}
+
+class _UsersPageState extends State<UsersPage> {
+  UsersRepository? _repository;
+
+  @override
+  void initState() {
+    super.initState();
+    _setupRepository();
+  }
+
+  Future<void> _setupRepository() async {
+    final client = await oauth2.clientCredentialsGrant(
+      Uri.parse(ApiConstants.ACCESS_TOKEN_URL),
+      ApiConstants.CLIENT_ID,
+      ApiConstants.CLIENT_SECRET,
+      scopes: ['read', 'write'],
+    );
+    _repository = UsersRepository(ApiClient(client));
+  }
+
+  Future<List<User>> _getUsers() async {
+    final users = await _repository?.getUsers();
```

Comment:
ok i have a few things i had to tweak here to get the list working. basically here is the issue.

![image](https://user-images.githubusercontent.com/5663734/122148973-aa2a8500-ce9e-11eb-9264-a50f7408c154.png)

now this happens because the call to configure/setup the repository is a future and that does not guarantee that it has finished setting up before this method has been called.

Reply:
That is correct, normally I would not create the repository there in the code. But since it was a small example, I just went for the fastest way of implementing this. Same for the UI, since there is no other code depending of these parts of the UI, I didn’t see the need of putting then into separate widgets.

Reply:
fair enough. although because of that i could not see the user list unless i create a new user as it's showing this error at start up.

Reply:
I mean you are right about that. I am still getting into the null-safety within dart specifically so I did see the error before but not consistently so I thought it had to do with something else. So thanks for pointing that out and clearing that up